lib/questions.rb

```diff
@@ -1,61 +1,82 @@
 # keep only the elements that start with an a
 def select_elements_starting_with_a(array)
+  array.select{|array| array[0] == "a"}
 end
 
 # keep only the elements that start with a vowel
 def select_elements_starting_with_vowel(array)
+  array.select {|x| ["a","e","i","o","u"].include? x[0]}
 end
 
 # remove instances of nil (but NOT false) from an array
 def remove_nils_from_array(array)
+  # array.delete(nil)
+  array.reject {|item| item == nil}
 end
 
 # remove instances of nil AND false from an array
 def remove_nils_and_false_from_array(array)
+    array.reject {|item| item == nil || item == false}
+
 end
 
 # don't reverse the array, but reverse every word inside it. e.g.
 # ['dog', 'monkey'] becomes ['god', 'yeknom']
 def reverse_every_element_in_array(array)
+  reverse_array = []
+  array.each{|x| reverse_array << x.reverse}
+  array = reverse_array
 end
 
 # given an array of student names, like ['Bob', 'Dave', 'Clive']
 # give every possible pairing - in this case:
 # [['Bob', 'Clive'], ['Bob', 'Dave'], ['Clive', 'Dave']]
-# make sure you don't have the same pairing twice, 
+# make sure you don't have the same pairing twice,
 def every_possible_pairing_of_students(array)
+  (array.combination(2).to_a).sort
 end
 
-# discard the first 3 elements of an array, 
+# discard the first 3 elements of an array,
 # e.g. [1, 2, 3, 4, 5, 6] becomes [4, 5, 6]
 def all_elements_except_first_3(array)
+  return array.slice(3..array.length)
 end
 
 # add an element to the beginning of an array
 def add_element_to_beginning_of_array(array, element)
+  array.unshift(1)
 end
 
 # sort an array of words by their last letter, e.g.
 # ['sky', 'puma', 'maker'] becomes ['puma', 'maker', 'sky']
 def array_sort_by_last_letter_of_word(array)
+
 end
 
 # cut strings in half, and return the first half, e.g.
 # 'banana' becomes 'ban'. If the string is an odd number of letters
 # round up - so 'apple' becomes 'app'
 def get_first_half_of_string(string)
+  half =  (string.length/2.0).round
+  string = string.slice(0..(half-1))
 end
 
 # turn a positive integer into a negative integer. A negative integer
 # stays negative
 def make_numbers_negative(number)
+  number = (number.abs) * -1
+  number
 end
 
-# turn an array of numbers into two arrays of numbers, one an array of 
+# turn an array of numbers into two arrays of numbers, one an array of
 # even numbers, the other an array of odd numbers
 # even numbers come first
 # so [1, 2, 3, 4, 5, 6] becomes [[2, 4, 6], [1, 3, 5]]
 def separate_array_into_even_and_odd_numbers(array)
+  # odd_array = array.select{|number| number.odd?}
+  # even_array = array.select{|number| number.even?}
+  return array.partition {|number| number.even?}
```

Comment:
Redundant `return` detected.<br>Space between { and | missing.<br>Space missing inside }.<br>Pass `&:even?` as an argument to `partition` instead of a block.